### ISSUE TITLE
Redundant `exponentialRampToValueAtTime` calls cause click noises

### ISSUE THREAD

**Issue by aratama** [benji6/virtual-audio-graph#43]
Thanks for this great library. 

I found some problems in `AudioParam`. When virtual-audio-graph updates `AudioParam` by function such as `exponentialRampToValueAtTime` multiple times, redundant function calls causes click noise. I confirmed it in Chrome 74 and Firefox 66, Windows 10 and Android 8. My [SSCCE](https://github.com/aratama/vag-sscce) and [demo](https://elated-golick-d45fd7.netlify.com/):

```ts  
import createVirtualAudioGraph, * as V from 'virtual-audio-graph';

document.getElementById("start").addEventListener("click", () => {
    const virtualAudioGraph = createVirtualAudioGraph();
    const currentTime = virtualAudioGraph.currentTime;
    setInterval(() => {
        virtualAudioGraph.update({
            0: V.gain('output', {
                gain: [
                    ['setValueAtTime', 1, currentTime + 1],
                    ['exponentialRampToValueAtTime', 0.0001, currentTime + 2]
                ]
            }),
            1: V.oscillator(0, { stopTime: currentTime + 2 })
        });
    }, 100);
});
```

I also found a following quick fix using a flag like `audioNode[key].applied = true;`: 

```ts 
if (Array.isArray(param)) {
    if (this_2.params && !(0, _utils.equals)(param, this_2.params[key])) {
        audioNode[key].cancelScheduledValues(0);
        audioNode[key].applied = false;
    }
    if (!audioNode[key].applied) {
        audioNode[key].applied = true;
        var callMethod = function (_a) {
        var methodName = _a[0],
            args = _a.slice(1);

        return (_b = audioNode[key])[methodName].apply(_b, args);

        var _b;
        };

        Array.isArray(param[0]) ? param.forEach(callMethod) : callMethod(param);
    }

} else {
    audioNode[key].value = param;
}
```

I can make a pull request about it, but it looks so dirty. I have not deep-dived into the source codes of this library, so I have no good idea to fix it yet. I believe `AudioWorkletVirtualAudioNode` also have same issue.

**Comment by benji6** [comment]
Hi thanks for reporting the issue, sorry for not responding quickly, I haven't looked at the source code in a while either but I'll dig in and see if I can find a solution

**Comment by benji6** [comment]
Looks like this was the fix 761c529a710336dda4196cc2b55be8fb61ad8d14 feel free to reopen if it didn't work and thanks for the very detailed bug report!

**Comment by aratama** [comment]
Thank you for your quick response! 
I'll check it out later.
